src/server.rs

```diff
@@ -87,15 +87,52 @@ async fn on_websocket_upgrade(socket: WebSocket) {
     };
 }
 
+fn get_index_listing(uri_path: &str) -> String {
+    let is_root = uri_path == "/";
+    let path = ROOT.get().unwrap().join(&uri_path[1..]);
+    let entries = fs::read_dir(path).unwrap();
+    let mut entry_names = entries
+        .into_iter()
+        .filter_map(|e| match e {
+            Ok(entry) => {
+                let trailing = if entry.metadata().unwrap().is_dir() {
+                    "/"
+                } else {
+                    ""
+                };
+                Some(format!(
+                    "{}{}",
+                    entry.file_name().to_str().unwrap(),
```

Comment:
Create a variable `file_name` just like the `trailing` to reduce the format statement length, and then write them into the `{}`.

Reply:
Ended up taking a slightly different approach here; take a look and let me know what you think. Feels a little cleaner to me, but I'm open to other options.

Reply:
Also a good way 👍